core/src/main/scala/org/bykn/bosatsu/rankn/Infer.scala

```diff
@@ -669,25 +669,25 @@ object Infer {
                 _ <- infer.set((Type.Fun(varT, bodyT), region(term)))
               } yield TypedExpr.AnnotatedLambda(name, varT, typedBody, tag)
           }
-        case AnnotatedLambda(name, tpe, result, tag) =>
+        case AnnotatedLambda(name, annotatedArgTy, result, tag) =>
           expect match {
             case Expected.Check((expTy, rr)) =>
               for {
                 vb <- unifyFn(expTy, rr, region(term))
-                (varT, bodyT) = vb
-                typedBody <- extendEnv(name, varT) {
+                (argT, resT) = vb
+                typedBody <- extendEnv(name, argT) {
                     // TODO we are ignoring the result of subsCheck here
                     // should we be coercing a var?
```

Comment:
This comment just means the `coerce` is being ignored? if the `subsCheck` fails then it all fails, right?

Reply:
that's right. We are only using it to check for failure.